rest/applications.go

```diff
@@ -28,6 +28,9 @@ type Applications interface {
 
 	GetGuildCommandsPermissions(applicationID snowflake.ID, guildID snowflake.ID, opts ...RequestOpt) ([]discord.ApplicationCommandPermissions, error)
 	GetGuildCommandPermissions(applicationID snowflake.ID, guildID snowflake.ID, commandID snowflake.ID, opts ...RequestOpt) (*discord.ApplicationCommandPermissions, error)
+
+	GetApplicationRoleConnectionMetadataRecords(applicationID snowflake.ID, opts ...RequestOpt) ([]discord.ApplicationRoleConnectionMetadata, error)
```

Comment:
I think we can omit `Records` for these endpoints, idk why discord names them like this